docs/Concepts/index.md

```diff
@@ -0,0 +1,247 @@
+# LocalStack Concepts
+
+When you first start working on LocalStack, you will most likely start working on AWS providers, either fixing bugs or adding features. In that case, you probably care mostly about [Services](#service), and, depending on the service and how it interacts with the [Gateway](#gateway), also **custom request handlers** and edge **routes**.
+
+If you are adding new service providers, then you’ll want to know how [Plugins](#plugins) work, and how to expose a service provider as a [service plugin](#service-plugin). This guide will give you a comprehensive overview about various core architectural concepts of LocalStack.
+
+## AWS Server Framework (ASF)
+
+AWS is essentially a Remote Procedure Call (RPC) system, and ASF is our server-side implementation of that system. The principal components of which are:
+
+-   Service specifications
+-   Stub generation
+-   Remote objects (service implementations)
+-   Marshalling
+-   Skeleton
+
+### Service specifications
+
+AWS developed a specification language, [Smithy](https://awslabs.github.io/smithy/), which they use internally to define their APIs in a declarative way. They use these specs to generate client SDKs and client documentation. All these specifications are available, among other repositories, in the [botocore repository](https://github.com/boto/botocore/tree/develop/botocore/data). Botocore are the internals of the AWS Python SDK, which allows ASF to interpret and operate on the service specifications. Take a look at an example, [the `Invoke` operation of the `lambda` API](https://github.com/boto/botocore/blob/474e7a23d0fd178790579638cec9123d7e92d10b/botocore/data/lambda/2015-03-31/service-2.json#L564-L573):
+
+```json 
+	"Invoke":{
+    "name":"Invoke",
+    "http":{
+      "method":"POST",
+      "requestUri":"/2015-03-31/functions/{FunctionName}/invocations"
+    },
+    "input":{"shape":"InvocationRequest"},
+    "output":{"shape":"InvocationResponse"},
+    "errors":[
+      {"shape":"ServiceException"},
+    ...
```

Comment:
I don't know what this is, but the three dots cause some weird UI changes which are visible not only in the PR diff, but also in the rendered markdown. I think this (and other occurrences) should be changed for better rendering.

Reply:
Just our of curiosity: Do you know what caused this weird rendering?